tests/utils/chain.go

```diff
@@ -40,12 +46,14 @@ func GetChainHeadWithError(t *testing.T, node *Node) (*types.Header, error) {
 	return headerResponseToHeader(t, header), nil
 }
 
-// GetBlockHash calls the endpoint chain_getBlockHash to get the latest chain head
-func GetBlockHash(t *testing.T, node *Node, num string) (common.Hash, error) {
-	respBody, err := PostRPCWithRetry(ChainGetBlockHash, NewEndpoint(node.RPCPort), "["+num+"]", 5)
-	if err != nil {
-		return common.Hash{}, err
-	}
+// GetBlockHash calls the endpoint chain_getBlockHash to get the latest chain head.
+// It will block until a response is received or the context gets canceled.
+func GetBlockHash(ctx context.Context, t *testing.T, node *Node, num string) (common.Hash, error) {
```

Comment:
Not sure why we're adding `ctx` to all these methods.  I haven't seen a case where we have to cancel these requests bound to some other context.  The actual test code diff will be smaller, if we just create a context with cancel in this function, and handle it internally.  It should return an error if it's past a supplied timeout duration.

Reply:
See above comment: https://github.com/ChainSafe/gossamer/pull/2385#discussion_r833071646